infer/modules/train/extract/extract_f0_print.py

```diff
@@ -1,175 +1,151 @@
+        # Open the logfiles within each thread in order to have anything done here...
+        # This makes the code I/O bound and there's not a lot to do about it, i think
+        __logfile = open(logfile, "w+")
+
+        out_features = predictor.compute_f0(load_audio(inp_path, 16000))
+        out_coarse = 0  # TODO: add coarse f0
+        np.save(
+            feature_path,
+            out_features,
+            allow_pickle=False,
         )
-        return f0_coarse
-
-    def go(self, paths, f0_method):
-        if len(paths) == 0:
-            printt("no-f0-todo")
-        else:
-            printt("todo-f0-%s" % len(paths))
-            n = max(len(paths) // 5, 1)  # 每个进程最多打印5条
-            for idx, (inp_path, opt_path1, opt_path2) in enumerate(paths):
-                try:
-                    if idx % n == 0:
-                        printt("f0ing,now-%s,all-%s,-%s" % (idx, len(paths), inp_path))
-                    if (
-                        os.path.exists(opt_path1 + ".npy") == True
-                        and os.path.exists(opt_path2 + ".npy") == True
-                    ):
-                        continue
-                    featur_pit = self.compute_f0(inp_path, f0_method)
-                    np.save(
-                        opt_path2,
-                        featur_pit,
-                        allow_pickle=False,
-                    )  # nsf
-                    coarse_pit = self.coarse_f0(featur_pit)
-                    np.save(
-                        opt_path1,
-                        coarse_pit,
-                        allow_pickle=False,
-                    )  # ori
-                except:
-                    printt("f0fail-%s-%s-%s" % (idx, inp_path, traceback.format_exc()))
-
-
-if __name__ == "__main__":
-    # exp_dir=r"E:\codes\py39\dataset\mi-test"
-    # n_p=16
-    # f = open("%s/log_extract_f0.log"%exp_dir, "w")
-    printt(" ".join(sys.argv))
-    featureInput = FeatureInput()
-    paths = []
-    inp_root = "%s/1_16k_wavs" % (exp_dir)
-    opt_root1 = "%s/2a_f0" % (exp_dir)
-    opt_root2 = "%s/2b-f0nsf" % (exp_dir)
+        # np.save(
+        #     coarse_path,
+        #     out_coarse,
+        #     allow_pickle=False,
+        # )
+    except Exception as e:
+        __log(__logfile, f"Failed to compute f0 for - {inp_path}: {e}")
+
+
+def extract_features(
+    predictor: F0Predictor, expected_dir: str, is_half: bool, device: str
+) -> None:
+    """
+    Extract features
+
+    Args:
+        expected_dir: str - Directory to store extracted features
+        cores: int - Number of CPU cores to use
+        method_f0: str - F0 method to use { pm, harvest, dio, rmvpe }
+    """
+    if predictor is RMVPE:
+        predictor = predictor("assets/rmvpe/rmvpe.pt", is_half, device)
+    elif predictor is CRePE or predictor is FCPE:
+        predictor = predictor(device=device)
+    else:
+        predictor = predictor()

```

Comment:
The parameter is important and you cannot use the default one. You can get these parameters from the original code.
ex. hop_len = 160, sr = 16000, etc.

Reply:
fixed in 5d75a21